doc/md/_a3_Getting-Started-Audio.md

```diff
@@ -0,0 +1,253 @@
+
+DaisySeed hw;
+daisysp::Oscillator osc;
+
+void MyCallback(AudioHandle::InputBuffer in, 
+                AudioHandle::OutputBuffer out, 
+                size_t size) 
+{
+    for (size_t i = 0; i < size; i++)
+    {
+        // The oscillator's Process function synthesizes, and
+        // returns the next sample.
+        float sine_signal = osc.Process();
+        out[0][i] = sine_signal;
+        out[1][i] = sine_signal;
+    }
+}
+
+int main(void)
+{
+    hw.Init();
+    // We initialize the oscillator with the sample rate of the hardware
+    // this ensures that the frequency of the Oscillator will be accurate.
+    osc.Init(hw.AudioSampleRate());
+    hw.StartAudio(MyCallback);
+    while(1) {}
+}
+```
+
+And with that, we have sound!
+
+Now, working with the input isn't too much different.
+
+### Audio Input
+
+The input comes in the same way that the audio goes out, as arrays of floating point values.
+
+If we simply want to pass audio through the Daisy without doing anything we just need to copy the data from the input buffer to the output buffer.
+
+```cpp
+for (size_t i = 0; i < size; i++)
+{
+    out[0][i] = in[0][i];
+    out[1][i] = in[1][i];
+}
+```
+
+but what's the point in setting up all of this code to make something sound the same, let's use the oscillator we already added to make something interesting happen:
+
+```cpp
+for (size_t i = 0; i < size; i++)
+{
+    float sine_signal = osc.Process();
+    out[0][i] = in[0][i] * sine_signal;
+    out[1][i] = in[1][i] * sine_signal;
+}
+```
+
+Now, whatever you send through the daisy is going to get multiplied by the 100Hz sine signal. Sounds pretty funky, right!? Well, that's a type of amplitude modulation known as ring modulation, which is a topic for another day, but it goes to show that with very little code, and some curiousity we can start to make some interesting things happen.
+
+[DaisySP](https://github.com/electro-smith/DaisySP) is filled with other objects to filter, delay, bit crush, reverberate, and otherwise synthesize, and process sound, and can be a great stepping off point from here to making some interesting sounds.
+
+### Audio Data Packing
+
+On the daisy, there are two ways the audio data can be prepeared, "Interleaved" and "Non-Interleaved", the overall callback structure functions the same in either case, but the way the input and output data is packed is different in each one.
+
+Above, we looked at the non-interleaved format, where each channel had its own array of samples.
+
+Interleaved audio is instead one large array containing multiple channels of data. This data is interleaved so that there is a single sample of each channel before moving to the next sample. For example, with a block size of four, the full buffer would look like:
+
+```
+{ L0, R0, L1, R1, L2, R2, L3, R3 }
+```
+
+as opposed to the non-interleaved version of the same data, that could be represented as:
+
+```
+{ 
+    { L0, L1, L2, L3 },
+    { R0, R1, R2, R3 }
+}
+```
```

Comment:
Maybe switch the order?

> Above, we looked at the non-interleaved format, where each channel had its own array of samples, like this:
> ```
> { 
>     { L0, L1, L2, L3 },
>     { R0, R1, R2, R3 }
> }
> ```
> 
> Interleaved audio is instead one large array containing multiple channels of data. This data is interleaved so that there is a single sample of each channel before moving to the next sample. For example, with a block size of four, the full buffer would look like:
> 
> ```
> { L0, R0, L1, R1, L2, R2, L3, R3 }
> ```